### PR TITLE
Implement access token authentication

### LINKED ISSUE #4: Implement JWT authentication with access tokens

### PR DESCRIPTION
**Pull request by VladimirV99**
Closes #4

### CODE REVIEW

File: Security/Identity/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        [ProducesResponseType(typeof(UserDetails), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<UserDetails>> UpdateName([FromBody] UserChangeNameRequest userChangeNameRequest)
+        {
+            var email = User.FindFirstValue(ClaimTypes.Email);
+            var user = await _repository.GetUserByEmail(email);
+            if (user != null)
+            {
+                await _repository.ChangeUserName(user, userChangeNameRequest.FirstName, userChangeNameRequest.LastName);
+                return Ok(_mapper.Map<UserDetails>(user));
```
Comment (aleksakojadinovic) [inline Security/Identity/Controllers/UserController.cs:51]: early returnnnn

File: Security/Identity/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            var user = await _repository.GetUserByEmail(email);
+            if (user != null)
+            {
+                var changed = await _repository.ChangeUserPassword(user, request.CurrentPassword, request.NewPassword);
+                if (changed)
+                    return Ok();
+                else
+                    return Unauthorized();
+            }
+            return NotFound();
```
Comment (aleksakojadinovic) [inline Security/Identity/Controllers/UserController.cs:80]: opet early return

File: Security/Identity/Swagger/AuthOperationFilter.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                .GetCustomAttributes(true)
+                .OfType<AuthorizeAttribute>();
+
+            var methodAuthAttributes = context.MethodInfo
+                .GetCustomAttributes(true)
+                .OfType<AuthorizeAttribute>();
+
+            var authAttributes = classAuthAttributes.Concat(methodAuthAttributes).Distinct();
+            if (authAttributes.Any())
+            {
```
Comment (aleksakojadinovic) [inline Security/Identity/Swagger/AuthOperationFilter.cs:28]: aj nek bude i ovde early return :DD

**Review by aleksakojadinovic — CHANGES_REQUESTED** [review]

File: Security/Identity/Controllers/AdministratorController.cs [hunk 4]
```diff
@@ -44,5 +45,25 @@ public async Task<IActionResult> GetUserByEmail(string email)
...
+                if (user.Email == User.FindFirstValue(ClaimTypes.Email))
+                    return BadRequest("Administrator can't delete his account");
+                if((await _repository.GetUserRoles(user)).Contains(Roles.ADMINISTRATOR))
+                    return BadRequest("Can't delete account of administrator");
+                
+                await _repository.DeleteUser(user!);
+                return Ok();
+            }
+            return NotFound();
+        }
```
[the remark this answers is not in the document]
Comment (aleksakojadinovic) [reply Security/Identity/Controllers/AdministratorController.cs:71]: stavi ove greske u konstante :D

**Review by Hos1g4k1 — APPROVED** [review]

**Review by aleksakojadinovic — APPROVED** [review]

**Pull request merged**